runtime/fundamentals/node.md

```diff
@@ -56,6 +56,175 @@ error: Relative import path "os" not prefixed with / or ./ or ../
 Same hints and additional quick-fixes are provided by the Deno LSP in your
 editor.
 
+## CommonJS support
+
+CommonJS is a module system that predetes
+[ES modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules).
+While we firmly believe that ES modules are the future of JavaScript, there are
+millions of npm libraries that are written in CommonJS and Deno offers full
+support for them. Deno will automatically determine if a package is using
+CommonJS and makes it work seemlessly when imported:
+
+```js, title="main.js"
+import react from "npm:react";
+console.log(react);
+```
+
+```shell
+$ deno run -E main.js
+18.3.1
+```
+
+_`npm:react` is a CommonJS package. Deno provides a seemless way to import it._
+
+That said, Deno strongly encourages use of ES modules, and offers restriced
+support for CommonJS in user code. There are two ways to write and consume
+CommonJS code
+
+### Use `.cjs` extension
+
+If the file extension is `.cjs` Deno will treat this module as CommonJS.
+
+```js title="main.cjs"
+const express = require("express");
+```
+
+Deno does not look for `package.json` files and `type` option to determine if
```

Comment:
```suggestion
Deno does not look for `package.json` files or `type` options to determine if
```

Reply:
I believe this is actually correct - `type` is only specified in `package.json`. So you can't have `type` without `package.json` hence the "and"